splunk-quarkus/src/main/java/com/redhat/console/integrations/splunk/SplunkIntegration.java

```diff
@@ -94,30 +85,13 @@ private void configureHandler() throws Exception {
                 .setHeader("targetUrl", simple("${headers.metadata[url]}"))
                 .setHeader("timeIn", simpleF("%d", System.currentTimeMillis()))
 
+                .process(new TargetUrlValidator()) // validate the TargetUrl to be a proper url
+
                 //Set Authorization header
                 .setHeader("Authorization", simpleF("Splunk %s", "${headers.metadata[X-Insight-Token]}"))
 
-                // body is a JsonObject so converting to consumable object
-                // for the http producer
-                .marshal().json(JsonLibrary.Jackson)
-
-                .setProperty("eventsCount", jsonpath("$.events.length()"))
-
-                // loops over events in the original message
-                .loop(exchangeProperty("eventsCount")).copy()
-                // picks one Event from the original message
-                .process(eventPicker)
-
-                // Transform message to add splunk wrapper to the json
-                .transform().simple("{\"source\": \"eventing\", \"sourcetype\": \"Insights event\", \"event\": ${body}}")
-
-                // aggregate transformed messages and append them together
-                // aggregate by "metadata" header as it contains data unique per target splunk instance
-                .aggregate(header("metadata"), new EventAppender())
-                // use default of 10 threads to process aggregated records (that act as queue)
-                .parallelProcessing(true)
```

Comment:
@gwenneg you mentioned that the change here is intentional. I assume it is because of simplification to EventSplitter... My concern here is that this is the only mechanism for a concurrent processing within this route (handler). See also https://github.com/RedHatInsights/eventing-integrations/pull/120

I think we should switch the handler route to SEDA completely.

Reply:
SEDA could be a good replacement of the parallel processing indeed. I'm curious about the Kafka messages acking strategy though. Kafka messages from a given partition have to be processed sequentially from a single thread so I suppose they will be acked before SEDA kicks in. That's something I'd like to confirm.

I'll be busy with other priorities for the next ~10 days so don't expect much activity from me in this draft :smiley: